src/components/core/text-field/text-field.tsx

```diff
@@ -0,0 +1,451 @@
+import zxcvbn from 'zxcvbn';
+import { forwardRef, Fragment, InputHTMLAttributes, ReactNode, useState } from 'react';
+import { styled } from 'Styles/stitches.config';
+
+type InputTypes = 'text' | 'number' | 'email' | 'password' | 'tel' | 'textarea';
+type TWordCountProps = { count: number; max_length: number };
+type TPasswordStrengthProps = { user_input: string; disable_meter: boolean; dark: boolean };
+
+export type TextFieldProps = InputHTMLAttributes<HTMLInputElement | HTMLTextAreaElement> & {
+    inline_prefix_element?: ReactNode;
+    inline_suffix_element?: ReactNode;
+    show_character_limit?: boolean;
+    label?: string;
+    type?: InputTypes;
+    max_length?: number;
+    error?: string;
+    success?: string;
+    hint?: string;
+    disabled?: boolean;
+    readonly?: boolean;
+    dark?: boolean;
+};
+
+/* 
+    PasswordStrengthMeter - This designs the section that displays the strength of password input 
+*/
+const StyledPasswordMeterWrapper = styled('div', {
+    height: '0.25rem',
+    width: '100%',
+    variants: {
+        dark: {
+            true: { background: '$greyDark500' },
+            false: { background: '$greyLight300' },
+        },
+    },
+});
+const StyledPasswordMeter = styled(StyledPasswordMeterWrapper, {
+    transition: 'width 0.25s ease-in-out',
+});
+const PasswordStrengthMeter = ({ user_input, disable_meter, dark }: TPasswordStrengthProps) => {
+    const test_result: zxcvbn.ZXCVBNResult = zxcvbn(user_input);
+    const score: number = (test_result.score * 100) / 4;
+    const meter_color: any = Object.freeze({
+        0: dark ? '$greyDark500' : '$greyLight300',
+        1: dark ? '$redDark' : '$redLight',
+        2: dark ? '$yellowDark' : '$yellowLight',
+        3: dark ? '$greenDark' : '$greenLight',
+        4: dark ? '$greenDark' : '$greenLight',
+    });
+    const generatePasswordStrengthColor = () => {
+        return {
+            width: `${score}%`,
+            background: meter_color[test_result.score],
+            height: disable_meter ? '0' : '0.25rem',
+        };
+    };
+    return (
+        <StyledPasswordMeterWrapper dark={!!dark}>
```

Comment:
```suggestion
        <StyledPasswordMeterWrapper dark={dark}>
```
I would go against the use of the `double-bang` operator. We already have type definition for these props making it our intention to accept boolean only. It is the responsibility of the user of this library to comply to this. Doing it this way would be make it harder to read

Reply:
I will refactor this